libs/trustchain/src/HWDeviceProvider.ts

```diff
@@ -0,0 +1,90 @@
+import { UserRefusedOnDevice } from "@ledgerhq/errors";
+import { ApduDevice } from "@ledgerhq/hw-trustchain/ApduDevice";
+import Transport, { StatusCodes, TransportStatusError } from "@ledgerhq/hw-transport";
+import { crypto, device } from "@ledgerhq/hw-trustchain";
+import api from "./api";
+import { genericWithJWT } from "./auth";
+import { AuthCachePolicy, JWT, TrustchainDeviceCallbacks } from "./types";
+
+export class HWDeviceProvider {
+  private policy?: AuthCachePolicy;
+  private jwt?: JWT;
+  private hw?: ApduDevice;
+
+  constructor(policy?: AuthCachePolicy) {
+    this.policy = policy;
+  }
```

Comment:
As policy is optional, don't we want to have a default value for it?

Reply:
also, I think `AuthCachePolicy` is not something we should set at the creation time, it's something that depends on the usecase usually.

for instance, we use a specific AuthCachePolicy on the `CloudSyncSDK#notifications` method to force a refresh (because there, we need to be sure to have a fresh jwt for the listenNotifications):

```ts
  listenNotifications(
    trustchain: Trustchain,
    memberCredentials: MemberCredentials,
  ): Observable<number> {
    const getFreshJwt = (): Promise<JWT> =>
      this.trustchainSdk.withAuth(
        trustchain,
        memberCredentials,
        jwt => Promise.resolve(jwt),
        "refresh",
      );
    return api.listenNotifications(getFreshJwt, this.slug);
  }
  ```